oddchain/src/Oddchain.hs

```diff
@@ -668,3 +710,100 @@ instance ToCBOR OddTxError where
 
 instance ToCBOR OutOfBoundError where
   toCBOR = encodeInt . fromEnum
+
+--------------------------------------------------------------------------------
+-- FromCBOR instances
+--------------------------------------------------------------------------------
+
+instance FromCBOR (Header OddBlock) where
+  fromCBOR = do
+    D.decodeListLenOf 2
+    !theSignedPart <- fromCBOR
+    !theSignature  <- fromCBOR
+    pure $! OddHeader
```

Comment:
If you have bangs on the field of `OddHeader`, do you still need the bangs here and `$!`? I'm assuming the result of the `Decoder` is evaluated to WHNF.

Reply:
No. It's just the habit.